### PR TITLE
Use growNoCopy in some places

### PR DESCRIPTION
**Pull request by easyice**
Found some `grow` method can be change to `growNoCopy` when reading code.

### CODE REVIEW

File: lucene/core/src/java/org/apache/lucene/util/BytesRefBuilder.java [hunk 1]
```diff
@@ -60,6 +60,13 @@ public void grow(int capacity) {
     ref.bytes = ArrayUtil.grow(ref.bytes, capacity);
   }
 
+  /**
+   * Used to grow the builder without coping bytes. see {@link ArrayUtil#growNoCopy(byte[], int)}.
```
Comment (dweiss) [inline lucene/core/src/java/org/apache/lucene/util/BytesRefBuilder.java:64]:
```suggestion
   * Used to grow the builder without copying bytes. see {@link ArrayUtil#growNoCopy(byte[], int)}.
```

Comment (easyice) [reply lucene/core/src/java/org/apache/lucene/util/BytesRefBuilder.java:64]: Thank you @dweiss , sorry for the typo!

File: lucene/core/src/java/org/apache/lucene/util/fst/Util.java
[the remark this answers is not in the document]
Comment (easyice) [reply lucene/core/src/java/org/apache/lucene/util/fst/Util.java:1]: Nice find! Thank you @epotyom !

**Review by epotyom — APPROVED** [review]
Nice improvement, approved!

**Comment by easyice** [comment]
Thank you for reviewing!

**Comment by easyice** [comment]
@epotyom @dweiss Hi, can you help me to merge if it looks okay?

**Review by dweiss — COMMENTED** [review]
LGTM. Will merge shortly. Thank you.

**Review by dweiss — APPROVED** [review]

**Comment by dweiss** [comment]
I've backported to branch_9x as well.

**Pull request merged**